packages/xo-server/src/api/pool.mjs

```diff
@@ -129,7 +129,7 @@ listMissingPatches.resolve = {
 // -------------------------------------------------------------------
 
 export async function installPatches({ pool, patches, hosts }) {
-  const opts = { patches }
+  const opts = { patches, xsCredentials: this.apiContext.user.preferences.xsCredentials }
```

Comment:
I'm wondering if it makes sense to have this attached to an XO user, as it's mainly related to a host instead, and it may be useful for other XO admins to install patches.

What do you think?

Reply:
The XS credentials are not bound to a specific host but to a person. You find them in your account and you can use them to download updates for any host. So I think it makes more sense to bind them to a user.

Reply:
@olivierlambert Any opinion on this?

Reply:
The question is: do we want to make it per user or should it be shared?

We'll find a way to do it if it should be shared.

Reply:
I suppose it will be less common to have a user with different Citrix/XS accounts to manage different pools vs everyone behind the same account. But both solution are "technically" valid. Ideally, something for a user where the user can decide to "share it" willingly to any other admin can make the updates. But it seems complicated.

Maybe we should start with a shared solution (easy to use) and then maybe in the future (if requested), make it more per group or user.

Reply:
So any XO Admin would be able to change it in XO and it would be used automatically by anyone (not necessarily an XO Admin) who installs updates on a XS pool? That wouldn't be an issue?

Where should it be in the UI?

Reply:
Only admins can make updates (IIRC), right? That's fine to me. But if it's too complicated, we can start per user. I prefer something that's available ASAP.

Reply:
I agree with Pierre, this raise security issue with personal data being shared with other admins.

As it's also acceptable for @olivierlambert, let's start per user.

Reply:
Yes, only *pool* admins can install updates. But they're not necessarily *XO* admins.